test/e2e/util.go

```diff
@@ -679,7 +680,13 @@ func waitForNamespacesDeleted(c *client.Client, namespaces []string, timeout tim
 }
 
 func waitForServiceAccountInNamespace(c *client.Client, ns, serviceAccountName string, timeout time.Duration) error {
-	Logf("Waiting up to %v for service account %s to be provisioned in ns %s", timeout, serviceAccountName, ns)
+	w, err := c.ServiceAccounts(ns).Watch(api.SingleObject(api.ObjectMeta{Name: serviceAccountName}))
```

Comment:
Can I take a rain check on that since we don't have watch returning consistent errors?

It would look like:

```
func WatcherFunc func() (Interface, error)

var nowFn = time.Now

func RetryUntil(timeout time.Duration, fn WatcherFunc, conditions ...ConditionFunc) (*Event, error) {
	start := nowFn()
	var lastEvent *Event
	for {
		w, err := fn()
		if err != nil {
			// insert retry logic
			return lastEvent, err
		}
		now := nowFn()
		remaining := now.Sub(start)
		if remaining < 0 {
			return lastEvent, wait.ErrWaitTimeout
		}
		lastEvent, err = Until(remaining, w, conditions...)
		if err != nil {
			// insert retry logic
			return lastEvent, err
		}
		return lastEvent, nil
	}
}
```